x/auth/types/client_tx.go

```diff
@@ -26,102 +28,67 @@ func (s *StdTxBuilder) SetMsgs(msgs ...sdk.Msg) error {
 	return nil
 }
 
-// GetSignatures implements TxBuilder.GetSignatures
-func (s StdTxBuilder) GetSignatures() []sdk.Signature {
-	res := make([]sdk.Signature, len(s.Signatures))
-	for i, sig := range s.Signatures {
-		res[i] = sig
-	}
-	return res
-}
-
 // SetSignatures implements TxBuilder.SetSignatures
-func (s *StdTxBuilder) SetSignatures(signatures ...client.Signature) error {
+func (s *StdTxBuilder) SetSignatures(signatures ...signing.SignatureV2) error {
 	sigs := make([]StdSignature, len(signatures))
 	for i, sig := range signatures {
-		pubKey := sig.GetPubKey()
+		pubKey := sig.PubKey
 		var pubKeyBz []byte
 		if pubKey != nil {
 			pubKeyBz = pubKey.Bytes()
 		}
+
+		var sigBz []byte
+		var err error
+		if sig.Data != nil {
+			sigBz, err = SignatureDataToAminoSignature(legacy.Cdc, sig.Data)
+			if err != nil {
+				return err
+			}
+		}
+
 		sigs[i] = StdSignature{
 			PubKey:    pubKeyBz,
-			Signature: sig.GetSignature(),
+			Signature: sigBz,
 		}
 	}
 	s.Signatures = sigs
 	return nil
 }
 
-// GetFee implements TxBuilder.GetFee
-func (s StdTxBuilder) GetFee() sdk.Fee {
-	return s.Fee
+func (s *StdTxBuilder) SetFee(amount sdk.Coins) {
```

Comment:
`Fee` reference looks bit ambiguous. Can we either change it to `SetFeeAmount` or update `SetFee` to accept input type of `Fee` directly